docs/src/piccolo/testing/index.rst

```diff
@@ -119,19 +127,64 @@ Alternatively, you can run the migrations to setup the schema if you prefer:
 
 .. code-block:: python
 
-    import asyncio
     from unittest import TestCase
 
     from piccolo.apps.migrations.commands.backwards import run_backwards
     from piccolo.apps.migrations.commands.forwards import run_forwards
+    from piccolo.utils.sync import run_sync
 
     class TestApp(TestCase):
         def setUp(self):
-            asyncio.run(run_forwards("all"))
+            run_sync(run_forwards("all"))
 
         def tearDown(self):
-            asyncio.run(run_backwards("all", auto_agree=True))
+            run_sync(run_backwards("all", auto_agree=True))
 
         def test_app(self):
             # Do some testing ...
             pass
+
+-------------------------------------------------------------------------------
+
+Testing async code
+------------------
+
+There are a few options for testing async code using pytest.
+
+You can either call any async code using Piccolo's ``run_sync`` utility:
+
+.. code-block:: python
+
+    from piccolo.utils.sync import run_sync
+
+    async def get_data():
+        ...
+
+    def test_get_data():
+        rows = run_sync(get_data())
+        assert len(rows) == 1
+
+Alternatively, you can make your tests natively async.
+
+If you prefer using pytest's function based tests, then take a look at
+`pytest-asyncio <https://github.com/pytest-dev/pytest-asyncio>`_. Simply
+install it using ``pip install pytest-asyncio``, then you can then write tests
+like this:
+
+.. code-block:: python
+
+    async def test_select():
+        rows = await MyTable.select()
+        assert len(rows) == 1
+
+If you prefer class based tests, then have a look at :class:`IsolatedAsyncioTestCase <unittest.IsolatedAsyncioTestCase>`
+from Python's standard library. You can then write tests like this:
+
+.. code-block:: python
+
+    from unittest import IsolatedAsyncioTestCase
+
+    class MyTest(IsolatedAsyncioTestCase):
+        async def test_select(self):
+            rows = await MyTable.select()
+            assert len(rows) == 1
```

Comment:
@dantownsend You should add a disclaimer or warning that `unittest.IsolatedAsyncioTestCase` can only be used since Python version `3.8` and above ([docs](https://docs.python.org/3.8/library/unittest.html#unittest.IsolatedAsyncioTestCase)).
I already make same mistake in [Piccolo API PR](https://github.com/piccolo-orm/piccolo_api/pull/74)

Reply:
Good point - I'll change it.